src/protocol/helper.c

```diff
@@ -136,13 +136,10 @@ protocolLocalParam(ProtocolStorageType protocolStorageType, unsigned int hostIdx
         // Add the process id -- used when more than one process will be called
         kvPut(optionReplace, VARSTR(CFGOPT_PROCESS_STR), VARUINT(processId));
 
-        // Add the group default id
-        kvPut(
-            optionReplace,
-            VARSTRZ(cfgOptionName(protocolStorageType == protocolStorageTypeRepo ? cfgOptRepoDefault : cfgOptPgDefault)),
-            VARUINT(
-                cfgOptionGroupIdxToKey(
-                    protocolStorageType == protocolStorageTypeRepo ? cfgOptGrpRepo : cfgOptGrpPg, hostIdx)));
+        // Add the pg default. Don't do this for repos because the repo default should come from the user or the local should
+        // handle all the repos equally. Repos don't get special handling like pg primaries or standbys.
+        if (protocolStorageType == protocolStorageTypePg)
+            kvPut(optionReplace, VARSTRDEF(CFGOPT_PG_DEFAULT), VARUINT(cfgOptionGroupIdxToKey(cfgOptGrpPg, hostIdx)));
```

Comment:
So would this be if they specify --pg-default= on the command line? And if they do --reset-pg-default, then what?

Reply:
The answer should be "nothing happens" but in fact the user reset takes precedence over our set here.

I believe the answer is to ignore the user reset when we are setting the option explicitly in code, see 2d55cffefd238.